test/gtest/mha_find20.cpp

```diff
@@ -0,0 +1,395 @@
+                                           numTensors,
+                                           arguments.get(),
+                                           workspace.ptr(),
+                                           workspace.size()),
+                         miopenStatusSuccess);
+
+            std::cerr << "Solution executed." << std::endl;
+
+            TensorStructMap outputTensorResults;
+
+            // reading results from find 2.0 call and preparing structures for non find 2.0 outputs
+            // to compare with
+            for(const auto& id : output_ids)
+            {
+                const TensorStructPtr& tensorStructPtr = tensors[id];
+                tensorStructPtr->tensorFloat.data      = handle.Read<float>(
+                    tensorStructPtr->gpuBuffer, tensorStructPtr->tensorFloat.data.size());
+
+                TensorStructPtr& ptr = outputTensorResults[id];
+
+                ptr              = TensorStructPtr(new TensorStruct());
+                ptr->tensorFloat = tensorStructPtr->tensorFloat;
+                ptr->gpuBuffer   = handle.Write(ptr->tensorFloat.data);
+            }
+
+            std::cerr << "Run via solver infrastructure directly." << std::endl;
+            GetResultsWithoutFind20(handle, outputTensorResults, workspace, solver_id);
+            std::cerr << "Run via solver infrastructure executed!" << std::endl;
+
+            for(const auto& id : output_ids)
+            {
+                double error = miopen::rms_range(outputTensorResults[id]->tensorFloat.data,
+                                                 tensors[id]->tensorFloat.data);
+                const double tolerance = 1e-3;
+
+                EXPECT_TRUE(std::isfinite(error) && error <= tolerance)
+                    << "Outputs do not match each other. Error:" << error;
+            }
+        }
+
+        std::cerr << "Finished testing forward solution functions." << std::endl;
+    }
+
+    void TestRunSolutionsBackward(Handle& handle, const std::vector<miopenSolution_t>& solutions)
+    {
+        std::cerr << "Testing backward solution functions..." << std::endl;
+
+        std::cerr << "Finished testing backward solution functions." << std::endl;
+    }
+
+    void Finalize() { EXPECT_EQUAL(miopenDestroyProblem(problem), miopenStatusSuccess); }
+
+private:
+    bool IsDropoutTensorId(miopenTensorArgumentId_t id) const
+    {
+        return id == miopenTensorMhaDropoutProbability || id == miopenTensorMhaDropoutSeed ||
+               id == miopenTensorMhaDropoutOffset;
+    }
+
+    void CreateTensor(miopenTensorArgumentId_t id,
+                      unsigned int n = 1,
+                      unsigned int c = 1,
+                      unsigned int h = 1,
+                      unsigned int w = 1,
+                      bool generate  = true)
+    {
+        // TODO Unused for now
+        bool floatFlag = !IsDropoutTensorId(id);
+        tensors[id]    = TensorStructPtr(new TensorStruct(floatFlag));
```

Comment:
Probably?
```suggestion
        tensors[id]    = std::make_shared<TensorStruct>(floatFlag);
```
Or `make_unique`.